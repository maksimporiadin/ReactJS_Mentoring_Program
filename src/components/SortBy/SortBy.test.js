import 'jsdom-global/register';
import React from 'react';

import { SortBy } from './index';

import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

configure({adapter: new Adapter()});

describe('SortBy', () => {
    let wrapper;
    const defaultProps = {
        onChange() {},
    };

    beforeEach(()=>{
        jest.resetAllMocks();
    });

    it('is renders with release_date', () => {
        const props = {
            sortBy: 'release_date'
        };

        wrapper = shallow(<SortBy {...defaultProps} {...props} />)
        expect(wrapper).toMatchSnapshot();
    });

    it('is renders with vote_average', () => {
        const props = {
            sortBy: 'vote_average'
        }

        wrapper = shallow(<SortBy {...defaultProps} {...props} />)
        expect(wrapper).toMatchSnapshot();
    });
});