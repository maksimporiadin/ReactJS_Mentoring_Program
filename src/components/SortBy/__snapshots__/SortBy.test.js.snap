// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortBy is renders with release_date 1`] = `
<div
  className="sortBy"
  onChange={[Function]}
>
  <span
    className="sortBy-release-date"
  >
    SORT BY: 
  </span>
  <label
    className="btn btn-outline-primary sortBy-input"
    htmlFor="sort-release-date"
  >
    <input
      defaultChecked={true}
      id="sort-release-date"
      name="sort"
      type="radio"
      value="release_date"
    />
     
    release date
  </label>
  <label
    className="btn btn-outline-secondary sortBy-input"
    htmlFor="sort-rating"
  >
    <input
      defaultChecked={false}
      id="sort-rating"
      name="sort"
      type="radio"
      value="vote_average"
    />
     
    rating
  </label>
</div>
`;

exports[`SortBy is renders with vote_average 1`] = `
<div
  className="sortBy"
  onChange={[Function]}
>
  <span
    className="sortBy-release-date"
  >
    SORT BY: 
  </span>
  <label
    className="btn btn-outline-secondary sortBy-input"
    htmlFor="sort-release-date"
  >
    <input
      defaultChecked={false}
      id="sort-release-date"
      name="sort"
      type="radio"
      value="release_date"
    />
     
    release date
  </label>
  <label
    className="btn btn-outline-primary sortBy-input"
    htmlFor="sort-rating"
  >
    <input
      defaultChecked={true}
      id="sort-rating"
      name="sort"
      type="radio"
      value="vote_average"
    />
     
    rating
  </label>
</div>
`;
